### DISCUSSION TITLE
Convergence criteria for MLMG::BottomSolver::bicgstab

### DISCUSSION THREAD

**Discussion by ZiyangHuang** [AMReX-Codes/amrex#3540] [Q&A] [answered]
Greetings,

I have a question of convergence of the ```MLMG::BottomSolver::bicgstab```. The matrix I have is strictly diagonal dominant, and I know that the Gauss-Seidel iteration is guaranteed to converge. I wonder if ```MLMG::BottomSolver::bicgstab``` has the same property.

If I want to use the Gauss-Seidel iteration as the bottom solver, is the code below a correct way to do so?
```mlmg.setBottomSolver(MLMG::BottomSolver::smoother);```
where ```mlmg``` is an instance of ```MLMG```.

Thanks!

**Comment by WeiqunZhang** [comment, marked as answer]
I don't know if bicgstab is in theory guaranteed to converge for diagonally dominant matrix. I suspect it is not. In my experience, if the initial guess is not good enough and the matrix is big, it may not converge.

Yes, that's the way to use a smoother at the bottom.
